sympy/solvers/solveset.py

```diff
@@ -221,7 +221,7 @@ def _invert_complex(f, g_ys, symbol):
         if isinstance(g_ys, FiniteSet):
             exp_invs = Union(*[imageset(Lambda(n, I*(2*n*pi + arg(g_y)) +
                                                log(Abs(g_y))), S.Integers)
-                               for g_y in g_ys])
+                               for g_y in g_ys if g_y != S.Zero])
```

Comment:
To be clear, my above comment is related to the relative performance:

from the following using `0` looks 2 times faster, when x is actually zero, whereas `S.Zero` is marginally faster when x is not zero

```python
In [19]: x = S(0)

In [20]: timeit x != S.Zero
1000000 loops, best of 3: 889 ns per loop

In [21]: timeit x != 0
1000000 loops, best of 3: 446 ns per loop

In [16]: x = Symbol('x')

In [17]: timeit x != S.Zero
100000 loops, best of 3: 6.32 µs per loop

In [18]: timeit x != 0
100000 loops, best of 3: 7.19 µs per loop
```
Note: I am fine with both, but `0` looks faster than S.Zero & relative occurrence of the former in the codebase is way more than the latter.